tests/dragonfly/generic_test.py

```diff
@@ -83,3 +83,26 @@ async def task2(k, n):
     await asyncio.gather(
         task1("i1", 2), task1("i2", 3), task2("l1", 2), task2("l1", 2), task2("l1", 5)
     )
+
+
+@dfly_args({"port": 6377})
+async def test_arg_from_environ_overwritten_by_cli(df_local_factory):
+    os.environ["DFLY_port"] = "6378"
+    dfly = df_local_factory.create()
+    dfly.start()
+    client = aioredis.Redis(port="6377")
+    await client.ping()
+
+
+async def test_arg_from_environ(df_local_factory):
+    # Using keys_output_limit flag to test environment variables parsing
+    max_keys = 512
+    os.environ["DFLY_keys_output_limit"] = str(max_keys)
+    dfly = df_local_factory.create()
+    dfly.start()
+    client = aioredis.Redis()
+    pipe = client.pipeline()
+    batch_fill_data(pipe, gen_test_data(max_keys * 3))
+    await pipe.execute()
+    keys = await client.keys()
+    assert len(keys) in range(max_keys, max_keys + 512)
```

Comment:
Its a tricky flag you've chosen, I'd pick something like `requirepass` that can be tested very easily